zio-http/shared/src/main/scala/zio/http/Body.scala

```diff
@@ -568,27 +570,32 @@ object Body {
     override def asStream(implicit trace: Trace): ZStream[Any, Throwable, Byte] =
       ZStream.unwrap {
         ZIO.blocking {
-          for {
-            r <- ZIO.attempt {
+          ZIO.suspendSucceed {
+            try {
               val fs   = new FileInputStream(file)
               val size = Math.min(chunkSize.toLong, file.length()).toInt
 
-              (fs, size)
-            }
-            (fs, size) = r
-          } yield ZStream
-            .repeatZIOOption[Any, Throwable, Chunk[Byte]] {
-              for {
-                buffer <- ZIO.succeed(new Array[Byte](size))
-                len    <- ZIO.attempt(fs.read(buffer)).mapError(Some(_))
-                bytes  <-
-                  if (len > 0) ZIO.succeed(Chunk.fromArray(buffer.slice(0, len)))
-                  else ZIO.fail(None)
-              } yield bytes
+              Exit.succeed {
+                ZStream.repeatZIOChunkOption {
+                  ZIO.suspendSucceed {
```

Comment:
@kyri-petrou Are we sure this `ZIO.suspendSucceed` and the `ZIO.attempt(fs.close())` will be run on a blocking thread?